pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -70,7 +72,8 @@ func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Cont
 	c := &azureclaim.AzureClaim{}
 	_, _, err = p.ParseUnverified(token.OAuthToken(), c)
 	if err != nil {
-		return nil, err
+		a.log.Error(err)
+		return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalToken, "properties.servicePrincipalProfile", "The provided service principal generated an invalid token.")
```

Comment:
`azRefreshableAuthorizer.NewRefreshableAuthorizerToken`  is currently only used in the ARO Operator controllers:

[Preview Feature Controller](https://github.com/Azure/ARO-RP/blob/7447d1b01df6974a8069b265f85fa59d7b15a0c5/pkg/operator/controllers/previewfeature/previewfeature_controller.go#L88)
[Storage Account Controller](https://github.com/Azure/ARO-RP/blob/7447d1b01df6974a8069b265f85fa59d7b15a0c5/pkg/operator/controllers/storageaccounts/storageaccount_controller.go#L104)
[Subnets Controller](https://github.com/Azure/ARO-RP/blob/7447d1b01df6974a8069b265f85fa59d7b15a0c5/pkg/operator/controllers/subnets/subnets_controller.go#L108)

I don't think these will be called from customer facing API requests, so we shouldn't change this to be a CloudError, but we could potentially just return a cleaner "The provided service principal generated an invalid token" message?